lib/modules/manager/flux/common.ts

```diff
@@ -8,3 +12,87 @@ export const systemManifestHeaderRegex =
 export function isSystemManifest(file: string): boolean {
   return regEx(systemManifestFileNameRegex).test(file);
 }
+
+export function resolveHelmReleaseManifest(
+  resource: HelmRelease,
+  helmRepositories: HelmRepository[],
+  helmCharts: HelmChart[],
+): {
+  name: string;
+  dep: PackageDependency;
+  matchingRepositories: HelmRepository[];
+} | null {
+  if (resource.spec.chartRef) {
+    const chartRef = resource.spec.chartRef;
+    const helmChart = helmCharts.find(
+      (res) =>
+        res.kind === chartRef.kind &&
+        res.metadata.name === chartRef.name &&
+        res.metadata.namespace ===
+          (chartRef.namespace ?? resource.metadata?.namespace),
+    );
+    if (!helmChart) {
+      return null;
+    }
+    const chartName = helmChart.spec.chart;
+
+    const dep: PackageDependency = {
+      depName: chartName,
+      currentValue: helmChart.spec.version,
```

Comment:
wait, this won't work as intended when the `HelmChart` is in a different file than the `HelmRelease` because renovate tries to replace the version in the wrong file.

instead of resolving the chartRef of helm release, you should collect the helm chart as separate dependency.

Reply:
right, I missed this and tested it only in cases where it was located in the same file, where it worked. on second look, it's even apparent in the test case where the expectation for the `packageFile` references the wrong name due to that.

I'll update/rework the MR accordingly